tools/internal/chartfetch/cmd.go

```diff
@@ -0,0 +1,167 @@
+		Short: "Downloads and unpacks Helm charts to a given directory",
+		Long:  chartFetchDesc,
+		Args:  cobra.ExactArgs(1),
+
+		// Only print out usage info when user supplies -h/--help
+		SilenceUsage: true,
+
+		// Don't print errors, we do it ourselves using a logging library
+		SilenceErrors: true,
+
+		// Main body of the command
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return fetchChart(args[0], options)
+		},
+	}
+
+	cobraCommand.Flags().StringVarP(&options.Version, "version", "v", "VERSION", "version of the chart to download. Required.")
+	cobraCommand.Flags().StringVarP(&options.DownloadDir, "download-dir", "d", "path/to/download/to", "path where chart should be downloaded. Required.")
+	if err := cobraCommand.MarkFlagRequired("version"); err != nil {
+		return nil, err
+	}
+	if err := cobraCommand.MarkFlagRequired("download-dir"); err != nil {
+		return nil, err
+	}
+
+	return cobraCommand, nil
+}
+
+// fetchChart actual command logic. Check if the chart directory exists. If so, download it.
+func fetchChart(chart string, options *Options) error {
+	// if download directory exists or we get an error checking for it, return
+	if exists, err := checkDownloadDirExists(options.DownloadDir); exists || err != nil {
+		return err
+	}
+
+	lockOptions := flock.Options{
+		Path:          siblingPath(options.DownloadDir, ".lk", true),
+		RetryInterval: lockRetryInterval,
+		Timeout:       lockTimeout,
+	}
+
+	return flock.WithLock(lockOptions, func() error {
+		// check for download directory again to make sure that it doesn't exist.
+		// (someone else could have gotten the lock and made it in the mean time)
+		if exists, err := checkDownloadDirExists(options.DownloadDir); exists || err != nil {
+			return err
+		}
+
+		return shellRunner.Run(shell.Command{
+			Prog: "helm",
+			Args: []string{"fetch", chart, "--untar", "-d", options.DownloadDir},
+		})
+	})
+}
+
+// checkDownloadDir checks if the download directory exists
+// returns true if it does
+func checkDownloadDirExists(path string) (bool, error) {
+	if _, err := os.Stat(path); err == nil {
+		log.Info().Msgf("Download directory %s exists, won't download again", path)
+		return true, nil
+	} else if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	} else {
+		return false, fmt.Errorf("unexpected filesystem error: %v", err)
+	}
+}
+
+// siblingPath is a path utility function. Given a directory path,
+// it returns a path in the same parent directory, with the same basename,
+// with a user-supplied suffix. The new path may optionally be hidden.
+//
+// Eg.
+//  siblingPath("/tmp/foo", ".lk", true)
+// ->
+// "/tmp/.foo.lk"
+//
+// siblingPath("/tmp/foo", "-scratch", false)
+// ->
+// "/tmp/foo-scratch"
+func siblingPath(relpath string, suffix string, hidden bool) string {
+	cleaned := path.Clean(relpath)
+	parent := path.Dir(cleaned)
+	base := path.Base(cleaned)
+
+	prefix := ""
+	if hidden {
+		prefix = "."
+	}
```

Comment:
```suggestion
	var prefix string
	if hidden {
		prefix = "."
	}
```
This is super nit picky and no need to change if you don't want to but I think this may be a more standard way of achieving the same thing.